### PR TITLE
refactor(icon-button): convert enzyme tests to RTL

### PR DESCRIPTION
**Pull request by tomdavies73**
### Proposed behaviour

Converts all enzyme tests to RTL

### Current behaviour

Currently all tests are written in enzyme

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/icon-button/icon-button.test.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+      <Icon type="bin" />
+    </IconButton>
+  );
+
+  const iconButton = screen.getByRole("button", {
+    name: "alternative-aria-label",
+  });
+
+  expect(iconButton).toBeVisible();
+  expect(iconButton).toHaveAccessibleName("alternative-aria-label");
```
Comment (nuria1110) [inline src/components/icon-button/icon-button.test.tsx:38]: **comment (non-blocking):** I don't think the accessible name assertion is necessary as  `getByRole()` already queries by the accessible name - see [docs](https://testing-library.com/docs/queries/byrole#:~:text=You%20can%20query,up%20to%20you.)

File: src/components/icon-button/icon-button.test.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    <IconButton>
+      <Icon type="bin" />
+    </IconButton>
+  );
+
+  const iconButton = screen.getByRole("button", { name: "bin" });
+
+  expect(iconButton).toBeVisible();
+  expect(iconButton).toHaveAccessibleName("bin");
+});
```
Comment (nuria1110) [inline src/components/icon-button/icon-button.test.tsx:52]: Same comment as above

File: src/components/icon-button/icon-button.test.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+test("should render with an `Icon` child", () => {
+  render(
+    <IconButton aria-label="icon-button">
+      <Icon type="bin" />
+    </IconButton>
+  );
+
+  const iconButton = screen.getByRole("button", { name: "icon-button" });
+  const icon = screen.getByTestId("icon");
```
Comment (edleeks87) [inline src/components/icon-button/icon-button.test.tsx:19]: comment (non-blocking): you could use `within` here and batch 2 of the assertions into 1 if you want but it's fine how it is.

```
  const icon = within(iconButton).getByTestId("icon");
  
   expect(iconButton).toBeVisible();
   expect(icon).toBeVisible();

**Review by nuria1110 — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**